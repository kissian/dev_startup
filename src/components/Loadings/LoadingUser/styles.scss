@mixin nth-children($points...) {
  @each $point in $points {
      &:nth-child(#{$point}) {
          @content;
      }
  }
}


.flex-column.timelineLoding {
  padding: 15px;
  display: flex;
  flex-direction: column;
  height: 414px;
  border: 1px solid #eee;
  border-radius: 0;
  width: 100%;
  margin-bottom: 30px;

  .flex-row-center {
    display: flex;
    flex-direction: row;
    align-items: center;
    .loadingCircle {
      border-radius: 100%;
      width: 50px; height: 50px;
      animation: pulse 1s infinite ease-in-out;
      
    }
    .loadingLine {
      height: 10px;
      margin: 10px 0;
      animation: pulse 1s infinite ease-in-out;
      @include nth-children(1) {
        width: 30%;
      }
      @include nth-children(2) {
        width: 50%;
      }
    }
  }
  .loadingLine {
    height: 10px;
    margin: 10px 0;
    animation: pulse 1s infinite ease-in-out;
    @include nth-children(3) {
      width: 100%;
    }
    @include nth-children(4, 5) {
      width: 100%;
    }
  }
  .loadingBox {
    width: 100%;
    height: 200px;
    background-color: #eee;
    animation: pulse 1s infinite ease-in-out;
  }
}

@keyframes pulse {
  0% {
      background-color: rgba(165, 165, 165, 0.1)
  }
  50% {
      background-color: rgba(165, 165, 165, 0.3)
  }
  100% {
      background-color: rgba(165, 165, 165, 0.1)
  }
}