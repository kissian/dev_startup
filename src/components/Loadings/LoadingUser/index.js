import React from 'react';
import {
    Row,
} from 'react-bootstrap';
import './styles.scss';

const LoadingUser = () => (
    <Row className="LoadingUser">
        <div className="loadingTitle" />
        <Row>
            <div className="loadingCircle" />
            <div className="loadingLine" />
            <div className="loadingLine" />
        </Row>
        <Row>
            <div className="loadingCircle" />
            <div className="loadingLine" />
            <div className="loadingLine" />
        </Row>
        <Row>
            <div className="loadingCircle" />
            <div className="loadingLine" />
            <div className="loadingLine" />
        </Row>
        <Row>
            <div className="loadingCircle" />
            <div className="loadingLine" />
            <div className="loadingLine" />
        </Row>
        <Row>
            <div className="loadingCircle" />
            <div className="loadingLine" />
            <div className="loadingLine" />
        </Row>
    </Row>
);
export default LoadingUser;
