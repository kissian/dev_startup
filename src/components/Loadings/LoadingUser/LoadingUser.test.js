import React from 'react';
import { shallow } from 'enzyme';
import LoadingUser from './index';

describe('LoadingUser component', () => {
    let component = null;
    it('renders component', () => {
        component = shallow(<LoadingUser />);
    });
    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
