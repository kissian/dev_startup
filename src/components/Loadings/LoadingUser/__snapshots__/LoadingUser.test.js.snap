// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingUser component matches snapshot 1`] = `
<div
  className="LoadingUser"
>
  <div
    className="loadingTitle"
  />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="loadingCircle"
    />
    <div
      className="loadingLine"
    />
    <div
      className="loadingLine"
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="loadingCircle"
    />
    <div
      className="loadingLine"
    />
    <div
      className="loadingLine"
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="loadingCircle"
    />
    <div
      className="loadingLine"
    />
    <div
      className="loadingLine"
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="loadingCircle"
    />
    <div
      className="loadingLine"
    />
    <div
      className="loadingLine"
    />
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <div
      className="loadingCircle"
    />
    <div
      className="loadingLine"
    />
    <div
      className="loadingLine"
    />
  </Row>
</div>
`;
